src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -194,4 +201,35 @@ boolean isJava8(List<File> services) {
     }
     return java8Detected;
   }
+
+  private Map<String, String> getAllAppEngineWebXmlEnvironmentVariables(List<File> services) {
+    Map<String, String> allAppEngineEnvironment = Maps.newHashMap();
+    for (File serviceDirectory : services) {
+      Path appengineWebXml = serviceDirectory.toPath().resolve("WEB-INF/appengine-web.xml");
+      try (InputStream is = Files.newInputStream(appengineWebXml)) {
+        Map<String, String> appEngineEnvironment = AppEngineDescriptor.parse(is).getEnvironment();
+        if (appEngineEnvironment != null) {
+          checkAndWarnDuplicateEnvironmentVariables(
+              appEngineEnvironment, allAppEngineEnvironment, serviceDirectory.getName());
```

Comment:
Since you are parsing the `appengine-web.xml`, why not report the real service name rather than the directory?  The directory name may not correspond to the service name.